sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1250,7 +1252,40 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
    */
   override def visitTableName(ctx: TableNameContext): LogicalPlan = withOrigin(ctx) {
     val tableId = visitMultipartIdentifier(ctx.multipartIdentifier)
-    val table = mayApplyAliasPlan(ctx.tableAlias, UnresolvedRelation(tableId))
+    val properties = new util.HashMap[String, String]
+    if (ctx.asOf != null && ctx.asOf.version != null) {
+      properties.put(TableCatalog.PROP_VERSION, ctx.asOf.version.getText)
+    } else if (ctx.asOf != null && ctx.asOf.TIMESTAMP != null) {
+      val ts = ctx.asOf.timestamp.getText
+      if (ts.length == "'yyyy-MM-dd'".length) {
```

Comment:
how about we just use `org.apache.spark.sql.catalyst.util.DateTimeUtils#parseTimestampString` here and make sure the timestamp parsing logic is consistent in Spark SQL?

Also why should we to match `'yyyy-MM-dd'` here? 
Could 
```
select * from t as of "2019-01-29 00:37:58"
```
be a valid statement?

Reply:
Changed. Thanks for the suggestion.